www/templates/html/Peoplefinder/Developers.tpl.php

```diff
@@ -1,174 +1,119 @@
-                                        case "xml":
-                                            $code = "xml";
-                                            break;
-                                        default:
-                                            $code = "html";
-                                    }
-                                    ?>
-                                    <pre class="code">
-                                        <code class="<?php echo $code; ?>"><?php echo htmlentities($result); ?></code>
-                                    </pre>
-                                </li>
-                            </ul>
-                        </div>
-                         <?php
-                     }
-                     ?>
+            </ul>
+            <div class="wdn_tabs_content">
+                <?php foreach ($resource->getAvailableFormats() as $format): ?>
 
-                </div>
-            </li>
-        </ul>
-    </div>
-
-</div>
-<div class="col right">
-    <div id='resources' class="zenbox primary" style="width:200px">
-        <h3>Directory API</h3>
-        <p>The following is a list of resources for Directory.</p>
-        <ul>
-            <?php
-            foreach ($context->resources as $resource) {
-                echo "<li><a href='?view=developers&resource=$resource'>$resource</a></li>";
-            }
-            ?>
-        </ul>
+                    <?php
+                    $url = UNL_Peoplefinder::addURLParams($resource->getExampleURI(), array('format' => $format));
+                    if (substr($url, 0, 2) == '//') {
+                        $url = 'http:' . $url;
+                    }
+                    $method_name = 'get' . ucfirst($format) . 'Properties';
+                    ?>
+                    <div id="<?php echo $format; ?>">
+                        <pre><code>GET <?php echo $url; ?></code></pre>
+                        <?php if (count($resource->$method_name())): ?>
+                            <h2>Resource Properties</h2>
+                            <table class="zentable neutral">
+                            <thead><tr><th>Property</th><th>Description</th></tr></thead>
+                            <tbody>
+                                <?php foreach ($resource->$method_name() as $property => $description): ?>
+                                    <tr>
+                                      <td><?php echo $property ?></td>
+                                      <td><?php echo $description ?></td>
+                                    </tr>
+                                <?php endforeach; ?>
+                            </tbody>
+                        </table>
+                        <?php endif; ?>
+                        <h2>Response</h2>
+                        <?php
+                        //Get the output.
+                        if (!$result = file_get_contents($url)) {
+                            $result = "Error getting file contents.";
```

Comment:
It seems to me like that approach would incur more technical debt and open the door to more unexpected side effects. A `file_get_contents` request should never fail if the page was already loaded (in theory I guess some problems could happen), and it would mimic the request exactly as intended.